src/test/func/malloc/malloc.cc

```diff
@@ -138,6 +138,22 @@ void test_memalign(size_t size, size_t align, int err, bool null)
   check_result(size, align, p, err, null);
 }
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+void test_reallocarr(void* p, size_t nmemb, size_t size, int err, bool null)
+{
+  size_t old_size = 0;
```

Comment:
This is only ever nullptr from the calling context.
```suggestion
void test_reallocarr(size_t nmemb, size_t size, int err, bool null)
{
  void* p;
  size_t old_size = 0;
```
Alternatively, did you mean
```suggestion
void test_reallocarr(void*& p, size_t nmemb, size_t size, int err, bool null)
{
  size_t old_size = 0;
```